src/promptflow/promptflow/_core/run_tracker.py

```diff
@@ -170,10 +170,15 @@ def _flow_run_postprocess(self, run_info: FlowRunInfo, output, ex: Optional[Exce
 
     def _update_flow_run_info_with_node_runs(self, run_info):
         run_id = run_info.run_id
-        run_info.api_calls = self._collect_traces_from_nodes(run_id)
         child_run_infos = self.collect_child_node_runs(run_id)
         run_info.system_metrics = run_info.system_metrics or {}
         run_info.system_metrics.update(self.collect_metrics(child_run_infos, self.OPENAI_AGGREGATE_METRICS))
+        run_info.api_calls = {
+            "start_time": run_info.start_time,
+            "end_time": run_info.end_time,
+            "children": self._collect_traces_from_nodes(run_id),
+            "system_metrics": run_info.system_metrics
+            }
```

Comment:
Consider using `Tracer.push()` to create the outermost level of the trace.

i.e. We can add it around `FlowExecutor.exec_line()`, similar to what this PR does: https://github.com/augaria/promptflow/pull/1/files?w=1#diff-faa6c81d614b7e41b18a42a93139d961d92afa9aa9dd0b72cb6b7176d7541e69

Reply:
Thanks @zzn2 !
That's great comment and much better than my current one.

Reply:
After some test, I noticed that the current tracer is designed for node_run + tool
Node_run start tracing with `start_tracing` method and tool with `push_tool` method
Adding a root level would introduce large code change and the removing of guardians code 
```
@classmethod
    def start_tracing(cls, run_id, node_name: Optional[str] = None):
        current_run_id = cls.current_run_id()
        **if current_run_id is not None:
            msg = f"Try to start tracing for run {run_id} but {current_run_id} is already active."
            logging.warning(msg)
            return**
        tracer = cls(run_id, node_name)
        tracer._activate_in_context()
```
in [start_tracing](https://github.com/microsoft/promptflow/blob/main/src/promptflow/promptflow/_core/tracer.py) function, which would take some time to investigate the full consequence.

I'll do some more investigation and see how much code change it would take.

Can we merge the first version code and enable the UI feature first?